packages/flutter_tools/lib/src/commands/fuchsia_reload.dart

```diff
@@ -172,11 +172,12 @@ class FuchsiaReloadCommand extends FlutterCommand {
 
   // Find ports where there is a view isolate with the given name
   Future<List<int>> _filterPorts(List<int> ports, String viewFilter) async {
+    printTrace('Looing for view $viewFilter');
     final List<int> result = <int>[];
     for (FlutterView v in await _getViews(ports)) {
       final Uri addr = v.owner.vmService.httpAddress;
       printTrace('At $addr, found view: ${v.uiIsolate.name}');
-      if (v.uiIsolate.name.indexOf(viewFilter) == 0)
+      if (v.uiIsolate.name.contains(viewFilter))
```

Comment:
Shouldn't it be
```dart
if (v.uiIsolate.name == viewFilter)
```
or
```dart
if (v.uiIsolate.name.startsWith(viewFilter))
```
Due to https://github.com/flutter/engine/commit/f5371156d29b8cca3df05452f7e09f1314b1d9a1 we know the exact format.

This can potentially allow invalid matches.
`foo1bar$main-10000000` and `foo2bar$main-11000000` will both match the filter `bar$main-1`

Reply:
We don't know the exact name because it depends on the uri of the app on the device, which is not available here.